cartography/intel/aws/kms.py

```diff
@@ -120,138 +125,133 @@ def get_grants(key: Dict, client: botocore.client.BaseClient) -> List[Any]:
 
 
 @timeit
-def _load_kms_key_aliases(
-    neo4j_session: neo4j.Session,
-    aliases: List[Dict],
-    update_tag: int,
-) -> None:
-    """
-    Ingest KMS Aliases into neo4j.
+def transform_kms_aliases(aliases: List[Dict]) -> List[Dict]:
     """
-    ingest_aliases = """
-    UNWIND $alias_list AS alias
-    MERGE (a:KMSAlias{id: alias.AliasArn})
-    ON CREATE SET a.firstseen = timestamp(), a.targetkeyid = alias.TargetKeyId
-    SET a.aliasname = alias.AliasName, a.lastupdated = $UpdateTag
-    WITH a, alias
-    MATCH (kmskey:KMSKey{id: alias.TargetKeyId})
-    MERGE (a)-[r:KNOWN_AS]->(kmskey)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = $UpdateTag
+    Transform AWS KMS Aliases to match the data model.
+    Converts datetime fields to epoch timestamps for consistency.
     """
+    transformed_data = []
+    for alias in aliases:
+        transformed = dict(alias)
 
-    neo4j_session.run(
-        ingest_aliases,
-        alias_list=aliases,
-        UpdateTag=update_tag,
-    )
+        # Convert datetime fields to epoch timestamps
+        transformed["CreationDate"] = dict_date_to_epoch(alias, "CreationDate")
+        transformed["LastUpdatedDate"] = dict_date_to_epoch(alias, "LastUpdatedDate")
+
+        transformed_data.append(transformed)
+    return transformed_data
 
 
 @timeit
-def _load_kms_key_grants(
-    neo4j_session: neo4j.Session,
-    grants_list: List[Dict],
-    update_tag: int,
-) -> None:
+def transform_kms_keys(
+    keys: List[Dict], policy_data: Optional[Dict[str, Dict]] = None
+) -> List[Dict]:
     """
-    Ingest KMS Key Grants into neo4j.
-    """
-    ingest_grants = """
-    UNWIND $grants AS grant
-    MERGE (g:KMSGrant{id: grant.GrantId})
-    ON CREATE SET g.firstseen = timestamp(), g.granteeprincipal = grant.GranteePrincipal,
-    g.creationdate = grant.CreationDate
-    SET g.name = grant.GrantName, g.lastupdated = $UpdateTag
-    WITH g, grant
-    MATCH (kmskey:KMSKey{id: grant.KeyId})
-    MERGE (g)-[r:APPLIED_ON]->(kmskey)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = $UpdateTag
+    Transform AWS KMS Keys to match the data model.
+    Converts datetime fields to epoch timestamps for consistency.
+    Optionally includes policy analysis properties.
     """
+    transformed_data = []
+    for key in keys:
+        transformed = dict(key)
+
+        # Convert datetime fields to epoch timestamps
+        transformed["CreationDate"] = dict_date_to_epoch(key, "CreationDate")
+        transformed["DeletionDate"] = dict_date_to_epoch(key, "DeletionDate")
+        transformed["ValidTo"] = dict_date_to_epoch(key, "ValidTo")
+
+        # Add policy analysis if provided
+        if policy_data and key["KeyId"] in policy_data:
+            transformed.update(policy_data[key["KeyId"]])
+        else:
+            # Set defaults for policy analysis properties
+            transformed["anonymous_access"] = False
+            transformed["anonymous_actions"] = []
```

Comment:
Here I'd prefer to be explicit.

```python
perms = key["keyId"]  # using a direct access here because if the data doesn't have a keyId then that's a big big problem and we should fail
transformed["anonymous_access"] = policy_data[perms].get("anonymous_access")  # using a `.get()` here because (now with my other suggestion) it's possible for anonymous_access to be None. This indicates that we don't know what the permission is.
transformed["anonymous_actions"] = policy_data[perms].get("anonymous_actions")  # same thing here. Instead of empty list, we have a None to indicate that we don't know. 
```

The end result if we cannot determine anonymous_actions for a given KMS key  is that the `KMSKey` node  will not have that field.